### PR TITLE
Android stack layout cr100

### PR DESCRIPTION
**Pull request by samartnik**
Resolves https://github.com/brave/brave-browser/issues/21556

This PR moves back removed code for stack layout.

Changes moved from chromium:
https://github.com/chromium/chromium/commit/94a231f90d89f9af2996172fae508f9472d573cb

Remove unused rendering code in TabLayout
Follow up to https://chromium-review.googlesource.com/c/chromium/src/+/3388174
and continues to simplify the Tab rendering logic for phones.

Also deletes anything related to tilting/rotating and card backs for the
tabs.

Bug: 1287653

https://github.com/chromium/chromium/commit/0d0895af67ad844e5292c4640a3b0fffbdf44839

[Jank] Make LayerTitleCache Tablet-only.
Noticed while looking at a startup/jank issue that we were creating the
LayerTitleCache. Now that we only ever use the java tab-switcher on
Phones, I opted to nuke it entirely from non-tablet UI.

Along the way, I moved ownership entirely out of CompostorViewHolder and
into LayoutManagerChromeTablet.

Change-Id: Ifaff8a7fc6eb507d0b7edd688a493103a55d2730

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++. I didn't remove extensively chromium parts, I mostly looked at our code modifications and patches.

File: chromium_src/chrome/browser/android/compositor/layer/tab_layer.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  void SetTitle(DecorationTitle* title);                                       \
+  void InitStack(LayerTitleCache* layer_title_cache);                          \
+  void SetProperties
+
+#define brightness_                                 \
+  Dummy() { return 0; }                             \
+  scoped_refptr<cc::Layer> title_;                  \
+  scoped_refptr<cc::UIResourceLayer> close_button_; \
+  raw_ptr<LayerTitleCache> layer_title_cache_;      \
+  float brightness_
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply chromium_src/chrome/browser/android/compositor/layer/tab_layer.h:37]: Unless the order of declaration matter ...

**Comment by deeppandya** [comment]
Thank you so much @samartnik for fixing this issue. it looks great.

**Comment by bsclifton** [comment]
Closing for now (see comments in https://github.com/brave/brave-browser/issues/21556#issuecomment-1085229467)

New issue created for re-adding Stack View:
https://github.com/brave/brave-browser/issues/22028

We can consider using this code OR doing our own implementation. @anthonypkeane to prioritize

**Pull request closed without merging**